app/policies/nil_class_policy.rb

```diff
@@ -0,0 +1,13 @@
+class NilClassPolicy < ApplicationPolicy
+  class NilClassPolicy < ApplicationPolicy
```

Comment:
You have a definition of policy inside a definition of policy with the same name. This is logically incorrect.